hazelcast/src/test/java/com/hazelcast/nio/tcp/nonblocking/SelectorModeTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.nio.tcp.nonblocking;
+
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.*;
+
+/**
+ * Test SelectorMode
+ */
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class SelectorModeTest {
+
+
+    @Test
+    public void getConfiguredValue_whenSelectWithFix() throws Exception {
+        System.setProperty("hazelcast.io.selectorMode", "selectwithfix");
+        assertEquals(SelectorMode.SELECT_WITH_FIX, SelectorMode.getConfiguredValue());
+        System.setProperty("hazelcast.io.selectorMode", "");
```

Comment:
I guess you want to store the original value and restore it after test. in a finally block?